pkg/cfgstruct/utils.go

```diff
@@ -0,0 +1,156 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package cfgstruct
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/zeebo/errs"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+// applyDefaultHostAndPortToAddr applies the default host and/or port if either is missing in the specified address.
+func applyDefaultHostAndPortToAddr(address, defaultAddress string) (string, error) {
+	defaultHost, defaultPort, err := net.SplitHostPort(defaultAddress)
+	if err != nil {
+		return "", err
+	}
+
+	addressParts := strings.Split(address, ":")
+	numberOfParts := len(addressParts)
+
+	if numberOfParts > 1 && len(addressParts[0]) > 0 && len(addressParts[1]) > 0 {
+		// address is host:port so skip applying any defaults.
+		return address, nil
+	}
+
+	// We are missing a host:port part. Figure out which part we are missing.
+	indexOfPortSeparator := strings.Index(address, ":")
+	lengthOfFirstPart := len(addressParts[0])
+
+	if indexOfPortSeparator < 0 {
+		if lengthOfFirstPart == 0 {
+			// address is blank.
+			return defaultAddress, nil
+		}
+		// address is host
+		return net.JoinHostPort(addressParts[0], defaultPort), nil
+	}
+
+	if indexOfPortSeparator == 0 {
+		// address is :1234
+		return net.JoinHostPort(defaultHost, addressParts[1]), nil
+	}
+
+	// address is host:
+	return net.JoinHostPort(addressParts[0], defaultPort), nil
+}
+
+// PromptForSatelitte handles user input for a satelitte address to be used with wizards
+func PromptForSatelitte(cmd *cobra.Command) (string, error) {
+	_, err := fmt.Print(`
+	Pick satellite to use:
```

Comment:
This will be indented:
![image](https://user-images.githubusercontent.com/7313031/57302284-1d1b7c00-70db-11e9-9a66-805abbb19e28.png)